Grillbot/Repository/TeamSearchRepository.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Discord;
+using Grillbot.Repository.Entity;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace Grillbot.Repository
+{
+    public class TeamSearchRepository : RepositoryBase
+    {
+        public TeamSearchRepository(IConfiguration config) : base(config)
+        {
+        }
+
+        public async Task<IEnumerable<TeamSearch>> GetAllSearchesAsync()
+        {
+            return await Context.TeamSearch.ToArrayAsync();
+        }
+        
+        public async Task AddSearch(IUser user, IChannel channel, string message)
+        {
+            await Context.TeamSearch.AddAsync(new TeamSearch
+                {UserId = user.Id.ToString(), Message = message, Category = channel.Name});
+            await Context.SaveChangesAsync();
+        }
+        
+        public async Task RemoveSearch(int id)
+        {
+            var row = Context.TeamSearch.First(d => d.Id == id);
```

Comment:
Maybe use FirstOrDefault and check on null? LINQ First throws exception if record not found.

Reply:
But that situation shouldn't ever happen, I call the method after I am sure that it exists in the Db, but it might happen in some data race situation? :thinking:

Reply:
But it won't hurt anyway :man_shrugging: